packages/contracts/contracts/ChainlinkAdapter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {AggregatorV3Interface} from "./Dependencies/AggregatorV3Interface.sol";
+
+contract ChainlinkAdapter is AggregatorV3Interface {
+    uint8 public constant override decimals = 18;
+    uint256 public constant override version = 1;
+
+    /**
+     * @notice Maximum number of resulting and feed decimals
+     */
+    uint8 public constant MAX_DECIMALS = 18;
+
+    /// @notice PriceFeed always fetches current and previous rounds. It's ok to
+    /// hardcode round IDs as long as they are greater than 0.
+    uint80 public constant CURRENT_ROUND = 2;
+    uint80 public constant PREVIOUS_ROUND = 1;
+    int256 internal constant ADAPTER_PRECISION = int256(10 ** decimals);
+
+    /**
+     * @notice Price feed for (BTC / USD) pair
+     */
+    AggregatorV3Interface public immutable BTC_USD_CL_FEED;
+
+    /**
+     * @notice Price feed for (ETH / USD) pair
+     */
+    AggregatorV3Interface public immutable ETH_USD_CL_FEED;
+
+    int256 internal immutable BTC_USD_PRECISION;
+    int256 internal immutable ETH_USD_PRECISION;
+
+    constructor(AggregatorV3Interface _btcUsdClFeed, AggregatorV3Interface _ethUsdClFeed) {
+        BTC_USD_CL_FEED = AggregatorV3Interface(_btcUsdClFeed);
+        ETH_USD_CL_FEED = AggregatorV3Interface(_ethUsdClFeed);
+
+        require(BTC_USD_CL_FEED.decimals() <= MAX_DECIMALS);
+        require(ETH_USD_CL_FEED.decimals() <= MAX_DECIMALS);
+
+        BTC_USD_PRECISION = int256(10 ** BTC_USD_CL_FEED.decimals());
+        ETH_USD_PRECISION = int256(10 ** ETH_USD_CL_FEED.decimals());
+    }
+
+    function description() external view returns (string memory) {
+        return "BTC/ETH Chainlink Adapter";
+    }
+
+    function _min(uint256 _a, uint256 _b) private pure returns (uint256) {
+        return _a < _b ? _a : _b;
+    }
+
+    function _convertAnswer(int256 ethUsdPrice, int256 btcUsdPrice) private view returns (int256) {
+        return
+            (ethUsdPrice * BTC_USD_PRECISION * ADAPTER_PRECISION) /
+            (ETH_USD_PRECISION * btcUsdPrice);
+    }
+
+    function latestRound() external view returns (uint80) {
+        return CURRENT_ROUND;
+    }
+
+    function _getRoundData(
+        AggregatorV3Interface _feed,
+        uint80 _roundId
+    ) private view returns (int256 answer, uint256 updatedAt) {
+        uint80 latestRoundId = _feed.latestRound();
+        uint80 feedRoundId;
+        (feedRoundId, answer, , updatedAt, ) = _feed.getRoundData(
+            _roundId == CURRENT_ROUND ? latestRoundId : latestRoundId - 1
```

Comment:
M:
This can cause 0 values to be returned as flagged by Watchpug
This will revert on line 72

Reply:
It will still be subject to the data not found when phaseId changes during the upgrade procedure of CL

This is a latent concern that is fine to ship with, but worth always keeping in mind

Reply:
https://gist.github.com/GalloDaSballo/f7b84c2716e681e2660dccf20165bc43